tests/namespaced.rs

```diff
@@ -666,34 +674,9 @@ mod namespaced {
         .await
     }
 
-    #[tokio::test]
-    async fn malicious_calls_sharednode() -> anyhow::Result<()> {
```

Comment:
nit: might be a good test for Dedicated mode still? Not quite sure.

Reply:
I don't think so - https://github.com/istio/ztunnel/pull/1031#discussion_r1659425358 was the only place where we had a "malicious calls" rule for Dedicated, and we don't need that - so there aren't any Dedicated-specific rules to test anymore.

Given the many different ways in which `ztunnel` _could_ be deployed and used in `Dedicated` mode, I dunno if it even makes sense to have any restrictions in that mode, since for all of them I can imagine valid exceptions.